packages/third-parties/src/google/gtm.tsx

```diff
@@ -0,0 +1,68 @@
+'use client'
+// TODO: Evaluate import 'client only'
+import React from 'react'
+import Script from 'next/script'
+
+declare global {
+  interface Window {
+    dataLayer?: Object[]
+    [key: string]: any
+  }
+}
+
+type GTMParams = {
+  gtmId: string
+  dataLayer: string[]
+  dataLayerName: string
+  auth: string
+  preview: string
+}
+
+let currDataLayerName: string | undefined = undefined
+
+export function GoogleTagManager(props: GTMParams) {
+  const { gtmId, dataLayerName = 'dataLayer', auth, preview, dataLayer } = props
+
+  if (currDataLayerName === undefined) {
+    currDataLayerName = dataLayerName
+  }
```

Comment:
Consider adding a warning in dev if you later update the layer name to something else. Since you cannot modify the layer name after initialization this is erroneous but since the default layer continues to work I'd just make it a console error since it'll just work with the initial layer in production